subsys/net/ip/net_context.c

```diff
@@ -1346,6 +1346,11 @@ NET_CONN_CB(tcp_syn_rcvd)
 
 		net_tcp_print_recv_info("SYN", pkt, NET_TCP_HDR(pkt)->src_port);
 
+		if (net_tcp_get_state(tcp) != NET_TCP_LISTEN) {
+			NET_DBG("we cannot process 2 SYNs here simultaneously.");
+			retrun NET_DROP;
```

Comment:
What will happen if our syn-ack is lost in transit, then the peer will re-send the syn. With your patch we would discard the this re-sent syn instead of sending syn-ack again (done few lines below). To me this looks like the connection establishment would fail in this case, or is my analysis correct?